File: backend/src/main/java/com/Revature/RevStay/models/Booking.java
package com.Revature.RevStay.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "bookings")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(nullable = false)
    private User customer;

    @ManyToOne
    @JoinColumn(nullable = false)
    private Hotel hotel;

    @ManyToOne
    @JoinColumn(nullable = false)
    private Room room;

    @JoinColumn(nullable = false)
    private LocalDate checkIn;

    @JoinColumn(nullable = false)
    private LocalDate checkOut;

    @JoinColumn(nullable = false)
    private Double totalPrice;

    @JoinColumn(nullable = false)
    private Integer numGuests;

    @Enumerated(EnumType.STRING)
    private BookingStatus status;
}

File: frontend/RevStay/src/components/reservation/Booking.ts
import IRoom from '../../interfaces/IRoom.ts';
import IHotel from '../../views/hotel/IHotel.ts'

export default interface Booking {
    id: number,
    hotel: IHotel,
    room: IRoom,
    checkIn: string,
    checkOut: string,
    totalPrice: number,
    numGuests: number,
    status: string,
}
